### PR TITLE
DevIL: add new recipe

### PR DESCRIPTION
**Pull request by n-morales**
### Summary
Changes to recipe:  **devil/1.8.0**

#### Motivation
DevIL (Developer's Image Library) is an open source image loading library https://github.com/DentonW/DevIL

#### Details
Adds the devil library version 1.8.0

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by n-morales** [comment]
Do I still need to include a v1 tester?

**Comment by AbrilRBS** [comment]
Hi @n-morales thanks a lot for taking the time to add this recipe, we appreciate it. I've submited some changes to your branch, feel free to take a look. They resolve the compilation issues we were seeing on static builds, and add lcms support, as the upstream cmakelists is looking for it and we would have to explicitly disable it otherwise.

I've removed the LIB set from the generate function, which afaik had no effect and Conan was already handling correctly on its own. Let me know if you see anything else, happy to help :)

File: recipes/devil/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+sources:
+  # Newer versions at the top
+  "1.8.0":
+    url:
+    - "https://github.com/DentonW/DevIL/archive/refs/tags/v1.8.0.tar.gz"
+    sha256: "52129f247b26fcb5554643c9e6bbee75c4b9717735fdbf3c6ebff08cee38ad37"
+patches:
+  "1.8.0":
```
[the remark this answers is not in the document]
Comment (n-morales) [reply recipes/devil/all/conandata.yml:8]: `patches/1.8.0-jp2_fix.patch` is similar to other fixes I found while searching for this issue including:
https://github.com/DentonW/DevIL/pull/102/files
https://gitlab.archlinux.org/archlinux/packaging/packages/devil/-/blob/main/jasper.patch?ref_type=heads

The jpeg fix and the remove register I came up with, but it looks like someone else did the same (for register fix) in https://github.com/DentonW/DevIL/pull/109/files

**Comment by n-morales** [comment]
> Hi @n-morales thanks a lot for taking the time to add this recipe, we appreciate it. I've submited some changes to your branch, feel free to take a look. They resolve the compilation issues we were seeing on static builds, and add lcms support, as the upstream cmakelists is looking for it and we would have to explicitly disable it otherwise.
> 
> I've removed the LIB set from the generate function, which afaik had no effect and Conan was already handling correctly on its own. Let me know if you see anything else, happy to help :)

Thanks for adjusting it! I don't have a mac os machine to test on so I was relying on the conan ci. It looks like there are some windows issues; I do have access to a windows machine but likely won't have the time to patch until the weekend.

**Comment by AbrilRBS** [comment]
I've launched the CI with the new piepeline, let's see how it it goes (results now go into the checks tab of the PR!)

**Comment by jcar87** [comment]
> I've launched the CI with the new piepeline, let's see how it it goes (results now go into the checks tab of the PR!)

mmm this hook warning would need to be addressed:
```
install_name dir of these shared libs is not @rpath: libIL.dylib, libILU.dylib, libILUT.dylib
```
(unless they're runtime-loaded plugins or something)

**Comment by n-morales** [comment]
> > I've launched the CI with the new piepeline, let's see how it it goes (results now go into the checks tab of the PR!)
> 
> mmm this hook warning would need to be addressed:
> 
> ```
> install_name dir of these shared libs is not @rpath: libIL.dylib, libILU.dylib, libILUT.dylib
> ```
> 
> (unless they're runtime-loaded plugins or something)

I'm not really familiar with mac os development -- how should these be addressed?

File: recipes/devil/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        # Let Conan handle the shared/static build
+        replace_in_file(self, os.path.join(self.source_folder, "DevIL", "src-ILU", "CMakeLists.txt"),
```
Comment (uilianries) [inline recipes/devil/all/conanfile.py:88]: Should we patch it or just accept upstream will and provide only shared library?

Comment (uilianries) [reply recipes/devil/all/conanfile.py:88]: Understood, the project generate both libraries at same time.

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/devil/all/conanfile.py (hunk 2 above)
Comment (AbrilRBS) [reply recipes/devil/all/conanfile.py:88]: I think this is good as-is, letting Conan chose it on its side. If there are issues with this being static on static builds, we will gadly accept reports!

**Comment by nmm0** [comment]
> Sorry for the delay @n-morales, I didn't catch this being stopped by the rpath issue. I've now pushed the last necessary changes to get this merged, thanks again for taking the time to submit the PR in the first place :)

Thanks @AbrilRBS for your help getting this all fixed up!

**Comment by n-morales** [comment]
> Sorry for the delay @n-morales, I didn't catch this being stopped by the rpath issue. I've now pushed the last necessary changes to get this merged, thanks again for taking the time to submit the PR in the first place :)

Thanks @AbrilRBS for your help getting this all fixed up!

**Pull request merged**